File: erpnext/accounts/report/segment_wise_profit_csd/segment_wise_profit_csd.py
# Copyright (c) 2013, Frappe Technologies Pvt. Ltd. and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe

def execute(filters=None):

	company = filters.get('company')
	from_date = filters.get('from_date')
	to_date = filters.get('to_date')
	columns = [
		("ACCOUNT") + "::250",
        ("CSD") + "::250",
        ("JUICES") + "::200",
        ("WATER") + "::150",
        ("CandyConfectionary") + "::120",
        ("Concentrates") + "::120",
        ("19LTR") + "::180",
		("Other") + "::180"
    ]
	data = []
	data = frappe.db.sql(""" 
		SELECT `account`,
			SUM(IF (`segment`='CSD' , `account_value`,0)) as csd, 
			SUM(IF (`segment`='JUICES' , `account_value` ,0))as juices,
			SUM(IF (`segment`='WATER' , `account_value`,0))AS  water,
			SUM(IF (`segment`='Candy&Confectionary' , `account_value`,0)) AS  candyconfectionary,
			SUM(IF (`segment`='Concentrates' , `account_value` ,0)) AS concentrates,
			SUM(IF (`segment`='19 LTR.'  ,`account_value`,0) ) as 19ltr,
			SUM(IF (`segment`='Other' , `account_value` ,0)) as other
		FROM `tabAccount Segment Data`
		WHERE date BETWEEN '{from_date}' and '{to_date}' 
		AND company = '{company}'
			group by `account`
	""".format(from_date=from_date, to_date=to_date,company=company), as_dict=True)
	
	return columns, data

File: erpnext/accounts/report/segment_wise_profit_csd/segment_wise_profit_csd.js
// Copyright (c) 2016, Frappe Technologies Pvt. Ltd. and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Segment Wise Profit CSD"] = {
	"filters": [
		{
			fieldname:"from_date",
			reqd: 1,
			label: __("From Date"),
			fieldtype: "Date",
			default: frappe.datetime.get_today(),
			
		},
		{
			fieldname:"to_date",
			reqd: 1,
			default: frappe.datetime.month_end(),
			label: __("To Date"),
			fieldtype: "Date",
		},				
		{
			fieldname:"company",
			reqd: 1,
			label: __("Company"),
			fieldtype: "Select",
			options: ["Unit 5", "Unit 8", "Unit 11"]
		}

	]
};
